src/preprocess.js

```diff
@@ -56,6 +56,45 @@ ${diagramText}
   return JSON.stringify(diagramObject)
 }
 
+// @ts-check
+/**
+ * @param {string} diagramText
+ * @param {any} context
+ * @returns {string}
+ */
+module.exports.preprocessPlantUML = function (diagramText, context) {
+  // see: http://plantuml.com/en/preprocessing
+  const RegExInclude = /^(\s*)!(include(?:sub)?)\s+(.+?)(?:!(\w+))?$/i
+  const diagramLines = diagramText.split('\n')
+  const diagramProcessed = diagramLines.map(line => line.replace(
+    RegExInclude,
+    (match, ...args) => {
+      const prefix = args[0]
+      // const include = args[1].toLowerCase()
+      const url = args[2].trim()
+      // const sub = args[3]
+      let vfs = context.vfs
+      if (typeof vfs === 'undefined' || typeof vfs.read !== 'function') {
+        vfs = require('./node-fs')
+      }
+      let data = line
+      try {
+        data = prefix + vfs.read(url)

```

Comment:
`prefix` is the whitespaces before the `include` directive? I'm not sure we need to preserve them but we will need to read the PlantUML code base to know:

1. if whitespaces are significant (I believe the answer is no)
2. if whitespaces are preserved or not when PlantUML resolves `include` directives.

I also assume that the `include` directive cannot be used inline.

Reply:
Yes it's before the `include` directive. You are right `include` directive cannot be used inline. I think whitespaces are not significant, i preserve them because my tests fail as the compared strings are different - so it's just easier to preserve them :)

Reply:
removed the whitespaces (prefix)